tests/unit_tests/test_data_structure.py

```diff
@@ -0,0 +1,179 @@
+            assert report.get_rule_results_stats()
+
+
+@pytest.mark.parametrize("to_remove, result", [
+    (
+        (),
+        {"low": 33, "medium": 351, "high": 25, "unknown": 33, "sum_of_filed_rules": 442}
+    ),
+    (
+        ("low", "medium", "high", "unknown"),
+        {"raise": True}
+    ),
+    (
+        ("fail"),
+        {"raise": True}
+    ),
+    (
+        ("low"),
+        {"low": 0, "medium": 351, "high": 25, "unknown": 33, "sum_of_filed_rules": 409}
+    ),
+    (
+        ("medium"),
+        {"low": 33, "medium": 0, "high": 25, "unknown": 33, "sum_of_filed_rules": 91}
+    ),
+    (
+        ("high"),
+        {"low": 33, "medium": 351, "high": 0, "unknown": 33, "sum_of_filed_rules": 417}
+    ),
+    (
+        ("unknown"),
+        {"low": 33, "medium": 351, "high": 25, "unknown": 0, "sum_of_filed_rules": 409}
+    ),
+    (
+        ("low", "medium"),
+        {"low": 0, "medium": 0, "high": 25, "unknown": 33, "sum_of_filed_rules": 58}
+    ),
+    (
+        ("high", "unknown"),
+        {"low": 33, "medium": 351, "high": 0, "unknown": 0, "sum_of_filed_rules": 384}
+    ),
+    (
+        ("medium", "high"),
+        {"low": 33, "medium": 0, "high": 0, "unknown": 33, "sum_of_filed_rules": 66}
+    ),
+    (
+        ("low", "unknown"),
+        {"low": 0, "medium": 351, "high": 25, "unknown": 0, "sum_of_filed_rules": 376}
+    ),
+    (
+        ("low", "medium", "high"),
+        {"low": 0, "medium": 0, "high": 0, "unknown": 33, "sum_of_filed_rules": 33}
+    ),
+    (
+        ("low", "medium", "unknown"),
+        {"low": 0, "medium": 0, "high": 25, "unknown": 0, "sum_of_filed_rules": 25}
+    ),
+    (
+        ("low", "high", "unknown"),
+        {"low": 0, "medium": 351, "high": 0, "unknown": 0, "sum_of_filed_rules": 351}
+    ),
+    (
+        ("medium", "high", "unknown"),
+        {"low": 33, "medium": 0, "high": 0, "unknown": 0, "sum_of_filed_rules": 33}
+    ),
+])
+def test_report_severity_of_failed_rules_stats(to_remove, result):
+    report = None
+    if "fail" in to_remove:
+        report = remove_all_rules_by_result(get_report(), to_remove)
+    else:
+        report = remove_all_rules_by_severity(get_report(), to_remove)
+
+    if "raise" not in result:
+        severity_of_failed_rules_stats = report.get_severity_of_failed_rules_stats()
+        print(severity_of_failed_rules_stats)
+        for key in result:
+            assert result[key] == severity_of_failed_rules_stats[key]
+    else:
+        with pytest.raises(MissingProcessableRules):
+            assert report.get_severity_of_failed_rules_stats()
+
+
+@pytest.mark.parametrize("system, type_of_remediation", [
+    ("Unknown_system", "Unknown_system"),
+    ("urn:xccdf:fix:script:sh", "Shell script"),
+    ("urn:xccdf:fix:script:ansible", "Ansible snippet"),
+    ("urn:xccdf:fix:script:puppet", "Puppet snippet"),
+    ("urn:redhat:anaconda:pre", "Anaconda snippet"),
+    ("urn:xccdf:fix:script:kubernetes", "Kubernetes snippet"),
```

Comment:
What about testing also some nonsense like what would happen if eg. "urn:xccdf:fix:script:pejskoskript" be passed?